api/controllers/build-log.js

```diff
@@ -9,74 +10,75 @@ function decodeb64(str) {
   return null;
 }
 
-module.exports = {
-  create: (req, res) => {
-    Promise.resolve(Number(req.params.build_id))
-    .then((id) => {
-      if (isNaN(id)) { throw 404; }
-      return Build.findByPk(id);
-    })
-    .then((build) => {
-      if (!build) {
-        throw 404;
-      } else if (build.token !== req.params.token) {
-        throw 403;
-      }
-
-      return BuildLog.create({
-        build: build.id,
-        output: decodeb64(req.body.output),
-        source: req.body.source,
-      });
-    })
-    .then(buildLog => buildLogSerializer.serialize(buildLog))
-    .then((buildLogJSON) => { res.json(buildLogJSON); })
-    .catch((err) => {
-      res.error(err);
+module.exports = wrapHandlers({
```

Comment:
Use `wrapHandlers` to have an easy way of handling errors when using `async` since they need to be explicitly caught. With this helper method, we don't have to remember to always add `try...catch` blocks. See the implementation for more info.